File: Data Structures & Algorithms/majority-element/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    majorityElement(nums) {
        let e = nums[0];
        let count = 1;
        let i = 1;
        while(i<nums.length){
            if(e===nums[i]){
                count++;
            }
            else{
                count--;
                if(count===0){
                    e=nums[i];
                    count=1;
                }
                
            }
            i++;
        }
        return e;
    }
}

File: Data Structures & Algorithms/search-insert-position/submission-4.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l = 0
        h = len(nums)-1
        while l<=h:
            m = (l + (h-l)//2)
            if(nums[m]==target):
                return m
            if target>nums[m]:
                l=m+1
            else:
                h=m-1;
        return l



        